ravenpy/utilities/geo.py

```diff
@@ -0,0 +1,238 @@
+      Path to output raster.
+    geometry : Union[Polygon, MultiPolygon, List[Union[Polygon, MultiPolygon]]
+      Geometry defining the region to crop.
+    touches : bool
+      Whether or not to include cells that intersect the geometry. Default: True.
+    fill_with_nodata: bool
+      Whether or not to keep pixel values for regions outside of shape or set as nodata. Default: True.
+    padded: bool
+      Whether or not to add a half-pixel buffer to shape before masking
+    raster_compression : str
+      Level of data compression. Default: 'lzw'.
+
+    Returns
+    -------
+    None
+    """
+    if not (type(geometry) in (list, tuple)):
+        geometry = [geometry]
+
+    with rasterio.open(raster, "r") as src:
+        mask_image, mask_affine = rasterio.mask.mask(
+            src,
+            geometry,
+            crop=True,
+            pad=padded,
+            all_touched=touches,
+            filled=fill_with_nodata,
+        )
+        mask_meta = src.meta.copy()
+        mask_meta.update(
+            {
+                "driver": "GTiff",
+                "height": mask_image.shape[1],
+                "width": mask_image.shape[2],
+                "transform": mask_affine,
+                "compress": raster_compression,
+            }
+        )
+
+        # Write the new masked image
+        with rasterio.open(output, "w", **mask_meta) as dst:
+            dst.write(mask_image)
+    return
+
+
+def generic_raster_warp(
+    raster: Union[str, Path],
+    output: Union[str, Path],
+    target_crs: Union[str, dict, CRS],
+    raster_compression: str = RASTERIO_TIFF_COMPRESSION,
+) -> None:
+    """
+    Reproject a raster file.
+
+    Parameters
+    ----------
+    raster : Union[str, Path]
+      Path to input raster.
+    output : Union[str, Path]
+      Path to output raster.
+    target_crs : str or dict
+      Target projection identifier.
+    raster_compression: str
+      Level of data compression. Default: 'lzw'.
+
+    Returns
+    -------
+    None
+    """
+    with rasterio.open(raster, "r") as src:
+        # Reproject raster using WarpedVRT class
+        with rasterio.vrt.WarpedVRT(src, crs=target_crs) as vrt:
+            # Calculate grid properties based on projection
+            affine, width, height = rasterio.warp.calculate_default_transform(
+                src.crs, target_crs, src.width, src.height, *src.bounds
+            )
+
+            # Copy relevant metadata from parent raster
+            metadata = src.meta.copy()
+            metadata.update(
+                {
+                    "driver": "GTiff",
+                    "height": height,
+                    "width": width,
+                    "transform": affine,
+                    "crs": target_crs,
+                    "compress": raster_compression,
+                }
+            )
+            data = vrt.read()
+
+            with rasterio.open(output, "w", **metadata) as dst:
+                dst.write(data)
+    return
```

Comment:
empty return like that? why?

Reply:
Oh, I think that's an artifact of my C++ background. Will remove!